addon/mixins/run.js

```diff
@@ -92,6 +92,54 @@ export default Mixin.create({
   },
 
   /**
+   Adds the provided function to the named queue to be executed at the end of the RunLoop.
+   The timer is properly canceled if the object is destroyed before it is invoked.
+
+   Example:
+
+   ```js
+   import Component from 'ember-component';
+   import ContextBoundTasksMixin from 'web-client/mixins/context-bound-tasks';
+
+   export default Component.extend(ContextBoundTasksMixin, {
+     init() {
+       this._super(...arguments);
+       this.scheduleTask('afterRender', () => {
+         console.log('This runs at the end of the run loop (via the afterRender queue) if this component is still displayed');
+       })
+     }
+   });
+   ```
+
+   @method scheduleTask
+   @param { Function } callback the callback to run at the provided time
+   @param { ...* } args arguments to pass to the callback
+   @param { Number } [timeout=0] the time in the future to run the callback
+   @public
+   */
+  scheduleTask(queue, callbackOrName, ...args) {
+    assert(`Called \`scheduleTask\` without a string as the first argument on ${this}.`, typeof queue === 'string');
+    assert(`Called \`scheduleTask\` on destroyed object: ${this}.`, !this.isDestroyed);
+    let type = typeof callbackOrName;
+
+    let cancelId = run.schedule(queue, this, () => {
+      let cancelIndex = this._pendingTimers.indexOf(cancelId);
+      this._pendingTimers.splice(cancelIndex, 1);
```

Comment:
`run.cancel` needs to do *something*, right? So it walks the list of all scheduled tasks (which may be long) and then must allocate a new array of valid tasks or mutate the existing array to remove the cancelled item.

Instead, it can be cheaper to allow a function to execute as scheduled, and simply make it a no-op at execution time. I looked at this approach for the `runTask` code, but I think in those cases avoiding the macro task queue item of a timeout makes cancellation worth the browser's cancellation expense. Here, I think the tradeoff is a bit different. Firing the scheduled task as a no-op is very cheap, cheaper than an extra iteration and allocation (or mutation) for cancellation.

The difference is between explicit cancellation:

```javascript
let tasks = [];
let task = { id: '123', fn: () => { /* do something */ } };
let cancelledTaskId = task.id;

tasks.push(task);

/* Now to cancel we must iterate and allocate a new `tasks` */
tasks = tasks.reject(t => t.id === cancelledTaskId);

/* Flush */
tasks.forEach(t => t.fn());
```

And lazy no-op:

```javascript
let tasks = [];
let taskCancelled = false;
let task = {
  id: '123',
  fn: () => {
    if (!taskCancelled) {
      /* do something */
    }
  }
};

tasks.push(task);

/* Now to cancel we change the value of the cancellation flag */
taskCancelled = true;

/* Flush */
tasks.forEach(t => t.fn());
```

Different implementations, the tests should be the same though.